### PR TITLE
[dv/top_level] Turn off ast_init_en overwrite by default

### LINKED ISSUE #18974: [dv/top_earlgrey] Remove setting ast_init_en
### Description

The SV code in chip_base_vseq, in function initialize_otp_creator_sw_cfg_ast_cfg, will overwrite ast_init_en in the otp_ctrl sw_config. There is a flag to control whether this is done, but that can only be set from SV code.

This code may have been written before the otp_ctrl config could be programatically generated, and it seems redundant nowadays. In fact, it gets on the way of running tests that need the ast NOT being initialized.

I think this code should be removed, and we should just rely on the otp images.

### PR DESCRIPTION
**Pull request by moidx**
This feature was added more than a year ago, and it seems redundant since the OTP image can control it. Perhaps it was a way to always run with calibrated clocks in simulation?

Fixes #18974

### CODE REVIEW

File: hw/top_earlgrey/dv/chip_sim_cfg.hjson [hunk 1]
```diff
@@ -442,22 +442,31 @@
       uvm_test_seq: chip_sw_all_escalation_resets_vseq
       sw_images: ["//sw/device/tests/sim_dv:all_escalation_resets_test:1"]
       en_run_modes: ["sw_test_mode_test_rom"]
-      run_opts: ["+bypass_alert_ready_to_end_check=1"]
+      run_opts: [
+        "+do_creator_sw_cfg_ast_cfg=1",
```
Comment (jdonjdon) [inline hw/top_earlgrey/dv/chip_sim_cfg.hjson:446]: Are you making this change for all tests?

Comment (matutem) [reply hw/top_earlgrey/dv/chip_sim_cfg.hjson:446]: I suppose we should need to set `do_creator_sw_cfg_ast_cfg=1` for tests that don't run rom_start.S, or have an OTP image that sets ast_init_en incorrectly. So it seems strange we need to set this plusargs for so many tests.

Comment (msfschaffner) [reply hw/top_earlgrey/dv/chip_sim_cfg.hjson:446]: I am not 100% sure, but I think @moidx just added it for all tests since they implicitly had it set before.
We probably still need to go through the exercise of removing the plusarg from tests that do not strictly require it.

File: hw/ip/clkmgr/data/clkmgr.hjson.tpl [hunk 2]
```diff
@@ -481,6 +481,7 @@
           '''
         },
       ]
+      tags: ["excl:CsrNonInitTests:CsrExclCheck"]
```
Comment (matutem) [inline hw/ip/clkmgr/data/clkmgr.hjson.tpl:484]: Is this intended? Seems unrelated.

File: hw/top_earlgrey/dv/chip_sim_cfg.hjson (hunk 1 above)
Comment (matutem) [reply hw/top_earlgrey/dv/chip_sim_cfg.hjson:446]: Two points:
- Please avoid adding "+do_creator_sw_cfg_ast_cfg=1" for each test, and simply add it to the list at line 229, which will affect all tests.
- Regardless of that, can we simply avoid setting that in the PR, run a full regression on the PR to identify the failures, and set the flag for the failures only? That way we will make progress.
